### ISSUE TITLE
Error while unit testing mocked dio object

### ISSUE THREAD

**Issue by TahaMalas** [flutterchina/dio#660]
### New Issue Checklist

- [X] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 
| ------------------------------ | -----------------------------------------------------
| Platform Name                                 |   flutter   |
| Platform Version                                        |   v1.12.13+hotfix.5   |
| Dio Version                                                     |  3.0.0    |
| Android Studio / Xcode Version        |   Android Studio 3.5 / Xcode 11.2.1      |

### Issue Description and Steps

The problem I am getting is that when I try to test a function that is calling a mocked dio object to make a POST request, this is the test I am trying to make

```
 test(
    'should perform a POST request on a URL with user/socialLogin being the endpoint and with application/json header',
    () async {
      // Arrange
      when(
        mockDio.post(
          any,
          options: OPTION,
          data: RequestBody.login(
            name: tName,
            email: tEmail,
            socialToken: tSocialToken,
            fcmToken: tFcmToken,
            image: tImage,
            language: tLanguage,
          ),
        ),
      ).thenAnswer((_) async =>
          Response(data: fixture('login_response.json'), statusCode: 200));
      // Act
      await remoteDataSource.login(
          tEmail, tName, tSocialToken, tFcmToken, tImage, tLanguage);
      // Assert
      verify(
        mockDio.post(
          Endpoints.LOGIN,
          options: OPTION,
          data: RequestBody.login(
            name: tName,
            email: tEmail,
            socialToken: tSocialToken,
            fcmToken: tFcmToken,
            image: tImage,
            language: tLanguage,
          ),
        ),
      );
    },
  );
```

The function I am trying to test is in the remoteDataSource and here is the implementation of the function

```
final formData = RequestBody.login(
        name: name,
        email: email,
        socialToken: socialToken,
        fcmToken: fcmToken,
        image: image,
        language: language);

await dio.post(Endpoints.LOGIN, options: OPTION, data: formData);
```

The main problem is in the form data object, because I tried to remove it from the function implementation and the verification process, and it worked, whereas when I run the test code above this problem occurs

```
No matching calls. All calls: MockDio.post(login, {data: Instance of 'FormData', queryParameters: null, options: Instance of 'Options', cancelToken: null, onSendProgress: null, onReceiveProgress: null})
(If you called `verify(...).called(0);`, please instead use `verifyNever(...);`.)

```

So basically I believe the test can not know how to compare to form data and he only uses reference equality, so the RequestBody.login() is returning a static object of the FormData I need 

```

class RequestBody {
  static FormData login(
      {@required String name,
      @required String email,
      @required String socialToken,
      @required String fcmToken,
      @required String image,
      @required String language}) {
    return FormData.fromMap({
      'name': name,
      'email': email,
      'socialToken': socialToken,
      'fcm_token': fcmToken,
      'image': image,
      'language': language,
    });
  }
}
```

And it is working well on version below 3.0.0, starting from this version and above the problem is occurring, any suggestions?

**Comment by TahaMalas** [comment]
0

**Comment by TahaMalas** [comment]
The problem is still occurring is there any fix for it?

**Comment by rohan20** [comment]
@TahaMalas I have the same issue. Asked on StackOverflow too: https://stackoverflow.com/questions/66549827/flutter-dio-test-url-endpoint-path

Were you able to solve this?

**Comment by joaoarmando** [comment]
@rohan20 same here! Did you get this?

**Comment by rohan20** [comment]
@joaoarmando No I wasn't 😞
